### PR TITLE
Allow updating textAlignVertical

### LINKED ISSUE #56933: _Decorator.updateRenderObject does not update textAlignVertical
I noticed this when a text field's vertical alignment wasn't updating on a setState call. It looks like the `_Decorator.updateRenderObject` method in `input_decorator.dart` doesn't update the `textAlignVertical` property, though it does update all other `_Decorator` properties . Assuming this was an unintentional omission?

### PR DESCRIPTION
**Pull request by justinmc**
## Description

`textAlignVertical` was not being updated in _Decorator.  This passes it through. 

## Related Issues

Closes https://github.com/flutter/flutter/issues/56933

## Tests

I added a regression test that updates the alignment from `top` to `bottom` and checks that the hint text has moved down.

## Breaking Change

No.

### CODE REVIEW

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -1707,10 +1707,14 @@ class _Decorator extends RenderObjectWidget {
...
      ..decoration = decoration
-     ..textDirection = textDirection
-     ..textBaseline = textBaseline
      ..expands = expands
-     ..isFocused = isFocused;
+     ..isFocused = isFocused
+     ..textBaseline = textBaseline
+     ..textDirection = textDirection;
+
+    if (textAlignVertical != null) {
```
Comment (justinmc) [inline packages/flutter/lib/src/material/input_decorator.dart:1715]: There is an assert to make sure that _RenderDecoration.textAlignVertical is not set to null, so I had to use this `if`.

**Review by emshack — COMMENTED** [review]
LGTM! But, not being on the Flutter team, I probably shouldn't be the one to give final approval :)

File: packages/flutter/lib/src/material/input_decorator.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/input_decorator.dart:1715]: Nit: it doesn't do anything if `textAlignVertical` is set to null from a non-null value? That sounds a little weird. `textAlignVertical` seems to be nullable but I couldn't find the corresponding behavior in the public documentation. Maybe we should document it?

**Review by LongCatIsLooong — APPROVED** [review]

**Comment by skia-gold** [comment]
Gold has detected about 1 untriaged digest(s) on patchset 3.
Please triage them at https://flutter-gold.skia.org/cl/github/57033

**Comment by justinmc** [comment]
@LongCatIsLooong Good point.  I've modified it so that it can be set to null now.  I think the assert was put there originally just to make the logic easier, and it was never used anyway because it wasn't possible to set textAlignVertical until now.

**Comment by justinmc** [comment]
The skia-gold comment was due to a typo that I pushed.  There should be no goldens change.

**Pull request merged**